server/controllers/userAuth.js

```diff
@@ -249,12 +250,15 @@ const resetPassword = (req, resp, next) => {
 };
 
 const updatePassword = (req, resp) => {
-  const { password: updatedPassword } = req.body;
+  const { password: updatedPassword, passwordConfirmation } = req.body;
   const { matches } = validator;
   const { token } = req.params;
   const sanitizedToken = sanitize(token);
+  const validationError = !matches(updatedPassword, /^[^\s]{4,32}$/);
+  const passwordsNoMatch =
+    _trim(updatedPassword) !== _trim(passwordConfirmation);
 
-  if (!matches(updatedPassword, /^[^\s]{4,32}$/)) {
+  if (validationError || passwordsNoMatch) {
     return resp.sendStatus(400);
```

Comment:
1. 406 status should be sent
2. You don't pass any errors to frontend. How frontend will know if the passwords don't match or are invalid. In the handleSubmit method in catch you get errors (line ~79) and save it to state, but they aren't sent from backend.